src/SensorNode.hpp

```diff
@@ -8,22 +8,35 @@ const char UNIT[] = "unit";
 class SensorNode : HomieNode
 {
 public:
-  SensorNode(const char* id);
+  SensorNode(const char* id, const char* unit);
   void setValue(const String& value);
+  bool setUnit(const char* unit);
 private:
   void onReadyToOperate();
+  char* node_unit;
+  bool unitSent;
 };
 
-SensorNode::SensorNode(const char* id): HomieNode(id, "sensor")
+SensorNode::SensorNode(const char* id, const char* unit): HomieNode(id, "sensor")
 {
+  node_unit = unit;
   advertise(VALUE);
   advertise(UNIT);
+  unitSent = false;
 }
 
 void SensorNode::onReadyToOperate(){
-  setProperty(UNIT).send("unit");
+  setProperty(UNIT).send(node_unit);
+  unitSent = true;
 }
 
 void SensorNode::setValue(const String& value){
   setProperty(VALUE).send(value);
 }
+
+
+void SensorNode::setUnit(const char* unit){
```

Comment:
Need to be `bool` not `void` also should the input be a String, as the pointer might be on the stack not the heap, depending where people call it ? Strings live on the heap (dynamic mem)
maybe add the homie logger message "can not set before bla ..."